cypress-tests/cypress/support/redirectionHandler.js

```diff
@@ -395,6 +395,15 @@ function upiRedirection(
 }
 
 function verifyReturnUrl(redirection_url, expected_url, forward_flow) {
+  const urlParams = new URLSearchParams(redirection_url.search);
+  const paymentStatus = urlParams.get('status');
+
+  // Check for valid statuses
+  if (paymentStatus !== 'succeeded' && paymentStatus !== 'processing' && paymentStatus !== 'requires_capture') {
+    throw new Error(`Payment failed after redirection with status: ${paymentStatus}`);
```

Comment:
There’s a possibility of receiving 5xx or 4xx errors during the redirection, which could result in the payment status being undefined. It would be helpful to print the error message in such cases.

Reply:
@Gnanasundari24 Done, review it now